libfreerdp/core/connection.c

```diff
@@ -618,16 +618,7 @@ BOOL rdp_client_redirect(rdpRdp* rdp)
 	settings = rdp->settings;
 	WINPR_ASSERT(settings);
 
-	if (settings->RedirectionFlags & LB_LOAD_BALANCE_INFO)
-	{
-		if (settings->LoadBalanceInfo && (settings->LoadBalanceInfoLength > 0))
-		{
-			if (!nego_set_routing_token(rdp->nego, settings->LoadBalanceInfo,
-			                            settings->LoadBalanceInfoLength))
-				return FALSE;
-		}
-	}
-	else
+	if (!(settings->RedirectionFlags & LB_LOAD_BALANCE_INFO))
```

Comment:
better us a `(flags & flag) == 0` syntax, `!` works from the standard, but I think it is more readable that way.

Reply:
ok, rebased your pr and changed the line.